pkg/storage/gc/gc.go

```diff
@@ -112,28 +119,39 @@ func (gc GarbageCollect) cleanRepo(repo string) error {
 	*/
 	index, err := common.GetIndex(gc.imgStore, repo, gc.log)
 	if err != nil {
+		gc.log.Error().Err(err).Str("module", "gc").Str("repository", repo).Msg("unable to read index.json in repo")
+
 		return err
 	}
 
 	// gc referrers manifests with missing subject and untagged manifests
-	if err := gc.cleanManifests(repo, &index); err != nil {
+	if err := gc.removeManifestsPerRepoPolicy(repo, &index); err != nil {
```

Comment:
Lines 128 should be after line 144?
First remove tags, and then remove all untagged manifests per policy. And at the end delete orphaned blobs.